core/trino-main/src/main/java/io/trino/operator/project/PageFilter.java

```diff
@@ -40,11 +38,9 @@ static SelectedPositions positionsArrayToSelectedPositions(boolean[] selectedPos
 
         int[] positions = new int[selectedCount];
         int index = 0;
-        for (int position = 0; position < size; position++) {
-            if (selectedPositions[position]) {
-                positions[index] = position;
-                index++;
-            }
+        for (int position = 0; index < selectedCount; position++) {
+            positions[index] = position;
```

Comment:
Add a comment that these writes are going to be overwritten (not intuitive)

Also, add a comment with the intent.